### PR TITLE
Add base class for camera settings providers

### PR DESCRIPTION
**Pull request by davidkline-ms**
This change adds a base class to handle the required properties and methods defined in the IMixedRealityCameraSettingsProvider interface.

The experimental UnityAR provider has been updated (with some other small changes) to use the base class.

### CODE REVIEW

File: Assets/MixedRealityToolkit.Extensions/Providers/Experimental/UnityAR/UnityARCameraSettings.cs [hunk 1]
```diff
@@ -37,38 +37,36 @@ public class UnityARCameraSettings : BaseDataProvider, IMixedRealityCameraSettin
...
-                QualitySettings.SetQualityLevel(cameraProfile.OpaqueQualityLevel, false);
-            }
-            else
-            {
-                CameraCache.Main.clearFlags = cameraProfile.CameraClearFlagsTransparentDisplay;
-                CameraCache.Main.backgroundColor = cameraProfile.BackgroundColorTransparentDisplay;
-                CameraCache.Main.nearClipPlane = cameraProfile.NearClipPlaneTransparentDisplay;
-                CameraCache.Main.farClipPlane = cameraProfile.FarClipPlaneTransparentDisplay;
-                QualitySettings.SetQualityLevel(cameraProfile.TransparentQualityLevel, false);
+                Debug.LogWarning("A profile was not specified for the XR Camera Settings provider.\nApplying Microsoft Mixed Reality Toolkit default options.");
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit.Extensions/Providers/Experimental/UnityAR/UnityARCameraSettings.cs]:
```suggestion
                Debug.LogWarning("A profile was not specified for the Unity AR Camera Settings provider.\nApplying Microsoft Mixed Reality Toolkit default options.");
```

Comment (keveleigh) [inline Assets/MixedRealityToolkit.Extensions/Providers/Experimental/UnityAR/UnityARCameraSettings.cs]: > Applying Microsoft Mixed Reality Toolkit default options.

Where does this happen?

Comment (davidkline-ms) [reply Assets/MixedRealityToolkit.Extensions/Providers/Experimental/UnityAR/UnityARCameraSettings.cs]: in ReadProfile... doesn't it?

Comment (davidkline-ms) [reply Assets/MixedRealityToolkit.Extensions/Providers/Experimental/UnityAR/UnityARCameraSettings.cs]: Oh. I see.... It's in the declarations of the variables. Is that a confusing pattern? I can change it.

File: Assets/MixedRealityToolkit.Extensions/Providers/Experimental/UnityAR/UnityARCameraSettings.cs [hunk 2]
```diff
@@ -37,38 +37,36 @@ public class UnityARCameraSettings : BaseDataProvider, IMixedRealityCameraSettin
...
-                QualitySettings.SetQualityLevel(cameraProfile.OpaqueQualityLevel, false);
-            }
-            else
-            {
-                CameraCache.Main.clearFlags = cameraProfile.CameraClearFlagsTransparentDisplay;
-                CameraCache.Main.backgroundColor = cameraProfile.BackgroundColorTransparentDisplay;
-                CameraCache.Main.nearClipPlane = cameraProfile.NearClipPlaneTransparentDisplay;
-                CameraCache.Main.farClipPlane = cameraProfile.FarClipPlaneTransparentDisplay;
-                QualitySettings.SetQualityLevel(cameraProfile.TransparentQualityLevel, false);
+                Debug.LogWarning("A profile was not specified for the Unity AR Camera Settings provider.\nApplying Microsoft Mixed Reality Toolkit default options.");
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit.Extensions/Providers/Experimental/UnityAR/UnityARCameraSettings.cs]:
```suggestion
                Debug.LogWarning("A profile was not specified for the Unity AR Camera Settings provider.\nUsing Microsoft Mixed Reality Toolkit default options.");
```

**Pull request merged**